scripts/publish-metadata.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import sys
+import requests
+import json
+import subprocess
+DESCRIPTION = """
+Uploads a given block of metadata to the Terra Data Catalog
+
+To run this script you must authed as an "admin" on the
+ specified environment's catalog resource
+
+"""
+
+def generate_access_token(user):
+    auth_as_user(user)
+
+    printProc = subprocess.Popen(
+        ["gcloud auth print-access-token"], stdout=subprocess.PIPE, shell=True
+    )
+    (out, err) = printProc.communicate()
+
+    return out.decode("ASCII").strip()
+
+
+def auth_as_user(user):
+    proc = subprocess.Popen(
+        [f"gcloud auth login {user} --brief"], stdout=subprocess.PIPE, shell=True
+    )
+    proc.communicate()
+
+
+def create_dataset(env, accessToken, storage_system, storage_system_id, entry):
+    headers = {
+        "Authorization": f"Bearer {accessToken}",
+        "Content-Type": "application/json",
+    }
+
+    request = {
+        "storageSystem": str(storage_system),
+        "storageSourceId": str(storage_system_id),
+        "catalogEntry": json.dumps(entry)
+    }
+    print(headers)
+    response = requests.post(
+        f"https://catalog.dsde-{env}.broadinstitute.org/api/v1/datasets", headers=headers, data=json.dumps(request)
+    )
+    print(response.text)
+    return None
+
+def main(args, current_user):
+    print(args)
+    access_token = generate_access_token(args.user)
+    print(access_token)
+    # Opening JSON file
+    with open(args.metadata_file, 'r') as f:
+        data = json.load(f)
+    create_dataset(args.environment, access_token, args.storage_system, args.storage_system_id, data)
+
+    auth_as_user(current_user)
```

Comment:
Another side effect I noticed was that we left you logged in as the account specified, which once again kinda causes changes to your system to be left behind. I thought it might be good to auth back when done. (Especially for folks who use their firecloud.org account in prod to not accidentally stay authed)